delivery_carrier_pricelist/models/delivery_carrier.py

```diff
@@ -0,0 +1,155 @@
+            rates = self.pricelist_send_shipping(pickings)
+            for index, rate in enumerate(rates):
+                result[index]["exact_price"] = rate["exact_price"]
+        return result
+
+    def _pricelist_get_price(self, order):
+        product_price = order.pricelist_id._get_product_price(
+            self.product_id,
+            1.0,
+            uom=self.product_id.uom_id,
+            date=order.date_order,
+        )
+        price = order.currency_id._convert(
+            product_price,
+            order.company_id.currency_id,
+            order.company_id,
+            order.date_order or fields.Date.today(),
+        )
+        return price
+
+    def pricelist_rate_shipment(self, order):
+        carrier = self._match_address(order.partner_shipping_id)
+        if not carrier:
+            return {
+                "success": False,
+                "price": 0.0,
+                "error_message": self.env._(
+                    "Error: this delivery method is not available for this address."
+                ),
+                "warning_message": False,
+            }
+        price = self._pricelist_get_price(order)
+        return {
+            "success": True,
+            "price": price,
+            "error_message": False,
+            "warning_message": False,
+        }
+
+    def pricelist_send_shipping(self, pickings):
+        res = []
+        for picking in pickings:
+            carrier = picking.carrier_id
+            sale = picking.sale_id
+            price = carrier._pricelist_get_price(sale) if sale else 0.0
+            res = res + [{"exact_price": price, "tracking_number": False}]
+        return res
+
+    def pricelist_get_tracking_link(self, picking):
+        return False
+
+    def pricelist_cancel_shipment(self, pickings):
+        raise NotImplementedError()
+
+    @api.model
+    def _get_view(self, view_id=None, view_type="form", **options):
+        arch, view = super()._get_view(
+            view_id=view_id, view_type=view_type, options=options
+        )
+        if view.name == "delivery.carrier.form":
+            arch = self._fields_view_get_adapt_attrs(arch)
+        return arch, view
+
+    @property
+    def attrs_list(self):
+        return ["invisible", "required", "readonly"]
+
+    def _add_pricelist_domain(
+        self,
+        doc,
+        xpath_expr,
+        attrs_key,
+        domain_operator="or",
+        field_operator="==",
+    ):
+        """Add the delivery type domain for 'pricelist' in attrs"""
+
+        if attrs_key not in self.attrs_list:
+            return
+
+        nodes = doc.xpath(xpath_expr)
+        for field in nodes:
+            domain = field.attrib.get(attrs_key, "")
+            if not domain:
+                continue
+
+            delivery_type_domain = f"delivery_type {field_operator} 'pricelist'"
+
+            domain = f"{domain} {domain_operator} {delivery_type_domain}"
+
+            field.set(attrs_key, domain)
```

Comment:
```suggestion
            delivery_type_domain = f"delivery_type {field_operator} 'pricelist'"
            domain = f"{domain} {domain_operator} {delivery_type_domain}"
            field.set(attrs_key, domain)
```
limit adding multiple blank lines in a method